cmd/argocd/commands/admin/backup.go

```diff
@@ -299,6 +317,24 @@ func NewImportCommand() *cobra.Command {
 					if !dryRun {
 						newLive := updateLive(bakObj, &liveObj, stopOperation)
 						_, err = dynClient.Update(ctx, newLive, metav1.UpdateOptions{})
+						if apierrors.IsConflict(err) {
+							fmt.Printf("failed to update %s/%s %s in namespace %s: %v\n", gvk.Group, gvk.Kind, bakObj.GetName(), bakObj.GetNamespace(), err)
+							if retryOnConflict {
+								err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+									fmt.Printf("resource conflict: retry update for Group: %s, Kind: %s, Name: %s, Namespace: %s\n", gvk.Group, gvk.Kind, bakObj.GetName(), bakObj.GetNamespace())
+									liveObj, getErr := dynClient.Get(ctx, newLive.GetName(), metav1.GetOptions{})
+									newLive.SetResourceVersion(liveObj.GetResourceVersion())
```

Comment:
Instead of using a retry, we should have a CLI option that removes resourceVersion before calling Update.

Reply:
kubenetes doesn't allow update when resourceVersion is empty. Good path would be to fetch the latest resourceVersion and then execute update.

Reply:
any concern, how i am updating the resource?

Reply:
```
applications.argoproj.io "dev-deploy-argocdephemeralaccess-usw2-qal" was not valid:
 metadata.resourceVersion: Invalid value: 0x0: must be specified for an update
```

Patch/update doesn't work with resourceversion set as empty